deeptime/markov/msm/markov_state_model.py

```diff
@@ -1087,77 +1100,49 @@ def score(self, dtrajs, score_method='VAMP2', score_k=10):
         """
-        from deeptime.markov.sample import ensure_dtraj_list
-        dtrajs = ensure_dtraj_list(dtrajs)  # ensure format
-        if self.count_model is None:
-            raise RuntimeError('This MarkovStateModel has not been estimated from data '
-                               '(e.g. count_model unassigned). Cannot proceed.')
-
-        # determine actual scoring rank
-        if score_k is None:
-            score_k = self.n_states
-        if score_k > self.n_states:
-            import warnings
-            warnings.warn('Requested scoring rank {rank} exceeds number of MSM states. '
-                          'Reduced to score_k = {n_states}'.format(rank=score_k, n_states=self.n_states))
-            score_k = self.n_states  # limit to n_states
-
-        # training data
-        K = self.transition_matrix  # model
-        C0t_train = self.count_model.count_matrix
-        from scipy.sparse import issparse
-        if issparse(K):  # can't deal with sparse right now.
-            K = K.toarray()
-        if issparse(C0t_train):  # can't deal with sparse right now.
-            C0t_train = C0t_train.toarray()
-        C00_train = np.diag(C0t_train.sum(axis=1))  # empirical cov
-        Ctt_train = np.diag(C0t_train.sum(axis=0))  # empirical cov
-
-        # test data
-        from deeptime.markov.tools.estimation import count_matrix
-        C0t_test_raw = count_matrix(dtrajs, self.count_model.lagtime, sparse_return=False)
-        # map to present active set
-        active_set = self.count_model.state_symbols
-        map_from = active_set[np.where(active_set < C0t_test_raw.shape[0])[0]]
-        map_to = np.arange(len(map_from))
-        C0t_test = np.zeros((self.n_states, self.n_states))
-        C0t_test[np.ix_(map_to, map_to)] = C0t_test_raw[np.ix_(map_from, map_from)]
-        C00_test = np.diag(C0t_test.sum(axis=1))
-        Ctt_test = np.diag(C0t_test.sum(axis=0))
-        assert C00_train.shape == C00_test.shape
-        assert C0t_train.shape == C0t_test.shape
-        assert Ctt_train.shape == Ctt_test.shape
-
-        # score
-        from deeptime.metrics import vamp_score
-        return vamp_score(K, C00_train, C0t_train, Ctt_train, C00_test, C0t_test, Ctt_test,
-                          k=score_k, score=score_method)
+        koopman_model = self.empirical_koopman_model
+        test_model = None
+        if dtrajs is not None:
+            from deeptime.markov.sample import ensure_dtraj_list
+            dtrajs = ensure_dtraj_list(dtrajs)  # ensure format
+            # test data
+            from deeptime.markov.tools.estimation import count_matrix
+            C0t_test_raw = count_matrix(dtrajs, self.count_model.lagtime, sparse_return=False)
+            # map to present active set
+            active_set = self.count_model.state_symbols
+            map_from = active_set[np.where(active_set < C0t_test_raw.shape[0])[0]]
+            map_to = np.arange(len(map_from))
```

Comment:
Maybe I'm not understanding this correctly, is this supposed to map between the active set of the self.count_model and the one of the C0t_test_raw?

Reply:
yes, i have updated the implementation using state symbols, i think it is more understandable and robust now.